Kernel/Arch/aarch64/RPi/InterruptController.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022, Timon Kruiper <timonkruiper@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+#include <Kernel/Arch/aarch64/IRQController.h>
+
+namespace Kernel::RPi {
+
+struct InterruptControllerRegisters;
+
+class InterruptController : public IRQController {
```

Comment:
This implements the RPi3 specific interrupt controller instead of the GIC that the Rpi4 supports, right?

In any case, maybe there could be a short overview comment about the situation of interrupt controllers on RPis and how this class fits in.

Reply:
Yes, I added a comment to file to clarify that!